src/Packlist/PacklistForm.js

```diff
@@ -0,0 +1,218 @@
+import React, { useState, useRef } from 'react'
+import { useForm } from 'react-hook-form'
+import { v4 as uuid } from 'uuid'
+import styled from 'styled-components/macro'
+import Input from '../common/Input'
+import Label from '../common/Label'
+import Checkbox from '../common/Checkbox'
+import ErrorMessage from '../common/ErrorMessage'
+import PropTypes from 'prop-types'
+import ListContainer from '../common/ListContainer'
+import ListItem from '../common/ListItem'
+import usePacklists from './usePacklists'
+
+PacklistForm.propTypes = {
+  onPacklistSave: PropTypes.func.isRequired,
+}
+
+export default function PacklistForm({ onPacklistSave }) {
+  const { register, handleSubmit, reset, errors } = useForm()
+  const [items, setItems] = useState([])
+  const { packlists } = usePacklists()
+  const itemRef = useRef(null)
+  const onSubmit = (packlist, event) => {
+    event.preventDefault()
+    // for testing...
+    if (event?.target && typeof event?.target.reset === 'function')
+      event.target.reset()
+    onPacklistSave({ name: packlist.name, packlist: items })
+  }
+  const handlePacklistKeyDown = (event) => {
+    if (event.key === 'Enter') {
+      event.preventDefault()
+      addItem({ text: event.target.value, id: uuid() })
+    }
+  }
```

Comment:
why do you decided to put these function above the return statement?

Reply:
because they are not working, when I put them after the return statement, I need the const for the return statement, so I need to declare them before the return statement starts